pkgs/tools/admin/pgadmin/update.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl wget jq yq common-updater-scripts prefetch-yarn-deps yarn-lock-converter
+
+set -eu -o pipefail
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+################################################################
+#         This script will update pgadmin4 in nixpkgs          #
+# Due to recent changes upstream, we will need to convert the  #
+#             `yarn.lock` file back to version 1.              #
+#   This isn't trivially done and relies on 3rd party tools    #
+#       and a hand-written converter (in this script).         #
+# Also, the converter cannot check for `github` repos in the   #
+#   `yarn.lock` file, which this script will add automatically #
+################################################################
+
+
+
+##################################################
+# Check for new version and download #
+##################################################
+
+scriptDir=$(cd "${BASH_SOURCE[0]%/*}" && pwd)
+nixpkgs=$(realpath "$scriptDir"/../../../..)
+
+newest_version="$(curl -s https://www.pgadmin.org/versions.json | jq -r .pgadmin4.version)"
+old_version=$(nix-instantiate --eval -E "(import \"$nixpkgs\" { config = {}; overlays = []; }).pgadmin4.version" | tr -d '"')
+url="https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v${newest_version}/source/pgadmin4-${newest_version}.tar.gz"
+
+if [[ $newest_version == $old_version ]]; then
+  printf "${GREEN}Already at latest version $newest_version${NC}\n"
+  exit 0
+fi
+printf "${YELLOW}New version: $newest_version ${NC}\n"
+
+# don't use mktemp, so if a network error happens, we can resume from there
+mkdir -p /tmp/pgadmin-update-script
+pushd /tmp/pgadmin-update-script
+wget -nc $url
+tar -xzf "pgadmin4-$newest_version.tar.gz"
+cd "pgadmin4-$newest_version/web"
+
+
+######################################
+# Convert the `yarn.lock` file to v1 #
+######################################
+
+printf "${YELLOW}Will now convert the v2 lockfile. This will download the npm packages to get the metadata.\n"
+printf "Please note: This will take some time!${NC}\n"
+yarn-lock-converter -i yarn.lock -o yarn_v1.lock --cache .cache
+printf "${GREEN}Conversion done${NC}\n"
+
+###########################################################################
+# Do some post-convert corrections (namely add '"' to multipackage lines) #
+###########################################################################
```

Comment:
```suggestion
# Do some post-convert corrections (namely add '"' to multipackage lines)
```
same here, I don't think the comments add to much value and we could remove them to shorten the entire script